frontend/src/modules/addonpromotion/components/AddonPromotion.js

```diff
@@ -0,0 +1,139 @@
+/* @flow */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Localized } from '@fluent/react';
+
+import './AddonPromotion.css';
+
+import * as user from 'core/user';
+
+import type { UserState } from 'core/user';
+
+type Props = {|
+    user: UserState,
+|};
+
+type InternalProps = {
+    ...Props,
+    dispatch: Function,
+};
+
+type State = {|
+    installed: boolean,
+|};
+
+/**
+ * Renders Pontoon Add-On promotion banner.
+ */
+export class AddonPromotionBase extends React.Component<InternalProps, State> {
+    constructor(props: InternalProps) {
+        super(props);
+        this.state = {
+            installed: false,
+        };
+    }
+
+    componentDidMount() {
+        window.addEventListener('message', this.handleMessages);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('message', this.handleMessages);
+    }
```

Comment:
If there are implemented with `useEffect`, this can be a nice functional component. ;)